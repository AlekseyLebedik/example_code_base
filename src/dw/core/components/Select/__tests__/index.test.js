import React from 'react';
import { shallow } from 'enzyme';

import Component from '../index';

describe('Select', () => {
  it('renders default values', () => {
    const props = {
      input: {
        value: null,
      },
      meta: {
        touched: undefined,
        error: undefined,
      },
    };
    expect(shallow(<Component {...props} />)).toMatchSnapshot();
  });

  it('renders custom values', () => {
    const props = {
      input: {
        value: 'item1',
        onChange: jest.fn(),
      },
      items: jest.fn(),
      meta: {
        touched: true,
        error: 'invalid',
      },
      label: 'Blah',
      fullWidth: true,
      multiple: true,
    };
    expect(shallow(<Component {...props} />)).toMatchSnapshot();
  });
});
