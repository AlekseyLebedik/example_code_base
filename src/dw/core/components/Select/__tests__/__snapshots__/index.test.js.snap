// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders custom values 1`] = `
<WithStyles(ForwardRef(TextField))
  SelectProps={
    Object {
      "MenuProps": Object {
        "PaperProps": Object {
          "style": Object {
            "maxHeight": 256,
          },
        },
        "anchorOrigin": Object {
          "horizontal": "left",
          "vertical": "bottom",
        },
        "getContentAnchorEl": null,
      },
      "classes": Object {
        "select": "select",
      },
      "multiple": true,
      "renderValue": [Function],
    }
  }
  classes={Object {}}
  disabled={false}
  fullWidth={true}
  input={
    Object {
      "onChange": [MockFunction],
      "value": "item1",
    }
  }
  items={[MockFunction]}
  label="Blah"
  meta={
    Object {
      "error": "invalid",
      "touched": true,
    }
  }
  onChange={[Function]}
  onDrop={[Function]}
  select={true}
  value={Array []}
/>
`;

exports[`Select renders default values 1`] = `
<WithStyles(ForwardRef(TextField))
  SelectProps={
    Object {
      "MenuProps": Object {
        "PaperProps": Object {
          "style": Object {
            "maxHeight": 256,
          },
        },
        "anchorOrigin": Object {
          "horizontal": "left",
          "vertical": "bottom",
        },
        "getContentAnchorEl": null,
      },
      "classes": Object {
        "select": "select",
      },
      "multiple": false,
      "renderValue": undefined,
    }
  }
  classes={Object {}}
  disabled={false}
  input={
    Object {
      "value": null,
    }
  }
  meta={
    Object {
      "error": undefined,
      "touched": undefined,
    }
  }
  onChange={[Function]}
  onDrop={[Function]}
  select={true}
  value=""
/>
`;
